### PR TITLE
Add feature for DOM Geometry

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds a feature for the Geometry Interfaces Module Level 1.

### CODE REVIEW

**Review by ddbeck — CHANGES_REQUESTED** [review]
I think the key question here is: what APIs actually return these types today?

I'm having a hard time reviewing this feature because it reads as extremely generic and I can't find any developers actually talking about "geometry level 1." As written, this feature tells a very unlikely story: that browsers shipped—and developers want—an abstract facility for doing geometry math.

The spec (and searching Stack Overflow) hints that this is not really case:

> The SVG interfaces [SVGPoint](https://drafts.fxtf.org/geometry/#svgpoint), [SVGRect](https://drafts.fxtf.org/geometry/#svgrect) and [SVGMatrix](https://drafts.fxtf.org/geometry/#svgmatrix) are aliasing the here defined interfaces in favor for common interfaces used by SVG, Canvas 2D Context and CSS Transforms. [[SVG11]](https://drafts.fxtf.org/geometry/#biblio-svg11) [[HTML]](https://drafts.fxtf.org/geometry/#biblio-html) [[CSS3-TRANSFORMS]](https://drafts.fxtf.org/geometry/#biblio-css3-transforms)

I don't know what the oldest thing is here but due to the limitations that we need to resolve in https://github.com/web-platform-dx/web-features/issues/971, I suspect these keys should (for now) reside with their oldest companion features, probably something like SVG or Canvas.

**Comment by dontcallmedom** [comment]
> I think the key question here is: what APIs actually return these types today?

see e.g. https://dontcallmedom.github.io/webidlpedia/names/DOMMatrix.html (section "Methods and attributes that return objects implementing DOMMatrix") - works for other names in the list as well.

> I'm having a hard time reviewing this feature because it reads as extremely generic and I can't find any developers actually talking about "geometry level 1.

+1

> I don't know what the oldest thing is here but due to the limitations that we need to resolve in #971, I suspect these keys should (for now) reside with their oldest companion features, probably something like SVG or Canvas.

[Historically](https://www.w3.org/TR/2013/WD-matrix-20130919/), 
> The transformation matrix interface replaces the SVGMatrix interface from SVG1.1

Among different usages, `SVGMatrix` was the type for the long-supported [ `SVGTransform.matrix`](https://developer.mozilla.org/en-US/docs/Web/API/SVGTransform)

File: features/geometry.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
+name: Geometry Level 1
```
Comment (Elchi3) [inline features/geometry.yml:1]:
```suggestion
name: Geometric shapes
```

File: features/geometry.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+description: The Geometry interfaces offer a way to represent points, rectangles, quadrilaterals and transformation matrices within JavaScript.
```
Comment (Elchi3) [inline features/geometry.yml:2]:
```suggestion
description: The `DOMMatrix`, `DOMPoint`, `DOMQuad` and `DOMRect` interfaces offer a way to represent points, rectangles, quadrilaterals and transformation matrices within JavaScript. They can be used in transformations in CSS, `<canvas>`, and SVG.
```

**Comment by Elchi3** [comment]
I've updated name and description here. I think this is part of the DOM and is an abstract set of interfaces that serve multiple other APIs such as CSS, canvas and SVG. Developers will encounter it in these contexts and if they have worked with it in say CSS, it will be familiar in SVG or canvas. I think it is fine to have such abstract features. Array is also abstract in that sense. I think Python doesn't have builtin matrices either and so numpy is an abstract package ("feature") which will let you work with matrices as a general thing. Does all this move us forward in some direction with this PR?

**Comment by Elchi3** [comment]
I discussed this with @ddbeck and we think landing this as "DOM Geometry" is best.

**Review by ddbeck — APPROVED** [review]
Thank you!

**Pull request merged**